ipxe/auto.go

```diff
@@ -140,6 +171,9 @@ func (h *Handler) defaultScript(span trace.Span, hw client.Discoverer, ip string
 func (h *Handler) customScript(hw client.Discoverer, ip string) (string, error) {
 	mac := hw.GetMAC(net.ParseIP(ip))
 	if chain := hw.Hardware().IPXEURL(mac); chain != "" {
+		if !strings.HasPrefix(chain, "http") && !strings.HasPrefix(chain, "https") {
```

Comment:
instead of checking it here, can this be done after `url.Parse` error checking and test against `u.Scheme`? that way, the dirty business of parsing urls is left to the url library.

Reply:
sure thing, will update.